content/articles/2019-04-dbplyr-1-4-0.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+
+Importantly, `lazy_frame()` has a `con` argument that allows you to specify which database should be used for the translation. This makes it easier to see the differences in SQL generation between databases:
+
+```{r}
+df1 %>% 
+  mutate(z = paste("item:", x)) %>% 
+  head(5)
+
+df2 <- lazy_frame(x = 1, y = 2, con = simulate_mssql())
+df2 %>% 
+  mutate(z = paste("item:", x)) %>% 
+  head(5)
+```
+
+When you don't specify a connection, dplyr uses its standard translation, which tries to follow the ANSI SQL standard as closely as possible.
+
+## Simpler SQL
+
+Two improvements have considerably reduced the number of subqueries that dbplyr needs:
+
+*   Joins, semi joins, and set operations no longer add additional unneeded
+    subqueries and now generate the minimum set:
+
+    ```{r}
+    df1 <- lazy_frame(x = 1, y = 2, a = 2)
+    df2 <- lazy_frame(x = 1, y = 2, b = 2)
+    
+    union(df1, df2)
+    
+    left_join(df1, df2, by = c("x", "y"))
+    
+    semi_join(df1, df2, by = c("x", "y"))
+    ```
+
+*   Many sequences of `mutate()`, `select()`, `rename()`, and `transmute()`
+    steps are be collapsed into a single query: 
+
+    ```{r}
+    df <- lazy_frame(x = 1, y = 2)
+    df %>% 
+      select(2:1) %>% 
+      select(2:1) %>% 
+      select(2:1) %>% 
+      select(2:1) %>% 
+      mutate(z = x + y) %>% 
+      select(3:1)
+    ```
+
+Note that dbplyr will still generate multiple subqueries from a single mutate statement when needed. This resolves one of my biggest frustrations with SQL:
+
+```{r}
+df %>% 
+  mutate(
+    a = x + 1, 
+    b1 = a * 3, 
+    b2 = a ^ 2,
+    c = b1 / b2
+  )
+```
+
+I touch on this advantage of dbplyr over SQL in a new vignette, [`vignette("sql")`](https://dbplyr.tidyverse.org/dev/articles/sql.html), which also gives some advice about how to write literal SQL, when dbplyr's built-in translations don't work.
+
+## SQL translation
+
+As well as improving the translation of high-level dplyr functions, we've also considerably added to the set of low-level vector functions that dbplyr can translate. Firstly, [MySQL][mysql-window] (>= 8.0), [MariaDB][maria-window] (>= 10.2) and [SQLite][sqlite-window] (>3.25) support for [window functions](https://dbplyr.tidyverse.org/dev/articles/translation-function.html#window-functions). These allow you to use summary functions (like `mean()` or `sum()`) inside of `mutate()`, as well as unlocking useful function like `min_rank()`, `first()`, and `lead()`/`lag()`.
+
+Thanks to [Cole Arendt](http://github.com/colearendt), dbplyr now supports translations for a selection of useful functions from stringr (`str_c()`, `str_sub()`, `str_length()`, `str_to_upper()`, `str_to_lower()`, and `str_to_title()`.) and lubridate ( `today()`, `now()`, `year()`, `month()` (numeric value only), `day()`, `hour()`, `minute()`, `second()`) :
```

Comment:
>  and `str_to_title()`.)

That period should go, and add a comma outside the parens before _and lubridate_.